### PR TITLE
Try: Make site editor responsive.

### PR DESCRIPTION
**Pull request by jasmussen**
The site editor is currently not usable on mobile:

before

This PR does a number of things to improve that:

- It fixes the white screen issue
- It hides pinned items entirely on mobile — these modal toggles are available in the More Options menu.
- It collapses the label, so it says "Update" instead of "Update Design", on mobile.
- It hides the same undo/redo/tools/block selector buttons on the site editor, that are hidden on the post editor. 

This improves things a fair bit:

Screenshot 2020-10-12 at 11 56 57

I wouldn't call this the end of the road, there's a great deal of polish that reains possible:

- The centered controls which are still in a design flux, probably shouldn't be optimized until we know how they will look like.
- There's some component usage that is treated differently based on post vs. site editor, this could be reduced and simplified.

But for now, this is a decent start.

### CODE REVIEW

**Comment by jasmussen** [comment]
This addresses feedback from https://github.com/WordPress/gutenberg/pull/25134#issuecomment-690629371.

File: packages/interface/src/components/interface-skeleton/style.scss
Comment (MichaelArestad) [inline packages/interface/src/components/interface-skeleton/style.scss:88]: I think this is really close. Something about this style is breaking the inspector panel on both the site editor and the post editor. You can see that here:

image

Comment (jasmussen) [reply packages/interface/src/components/interface-skeleton/style.scss:88]: Awesome catch, I'll need to address that! Thank you.

Comment (jasmussen) [reply packages/interface/src/components/interface-skeleton/style.scss]: Alright this should be fixed.

**Comment by jasmussen** [comment]
Alright this one should be ready to go.

It would be really nice to get this one in soon, both so the experience isn't totally broken on mobile, but also so that actually optimizing for and considering mobile can be easier as development moves on.

**Review by david-szabo97 — COMMENTED** [review]
The inserter panel and name of the panel are not showing up for me.

image

**Comment by jasmussen** [comment]
Argh how could I miss that !?

Taking a look, thank you!

**Comment by jeyip** [comment]
I just pushed a commit to address [this comment](https://github.com/WordPress/gutenberg/pull/25884#discussion_r511776113), which seemed to be causing layout issues for mobile viewports. The header at smaller viewport widths should now look like this:

Screen Shot 2020-10-27 at 1 58 21 AM

I pushed it here because I thought it would be easier to test, discuss, and potentially include with the fixes made in this pull request, but I'm more than happy to rebase and remove the commit and include it in separate PR.

**Comment by jasmussen** [comment]
Awesome.

There are a lot of good CSS changes in this PR that I believe will hold us over for a while wrt the mobile site editor. But it's still blocked on the white screen, and I think I'll need some help getting that working. Behold:

white

That white screen is actually the "left sidebar" being open, all the time.

**Comment by jeyip** [comment]
> There are a lot of good CSS changes in this PR that I believe will hold us over for a while wrt the mobile site editor. But it's still blocked on the white screen, and I think I'll need some help getting that working. Behold:

I pushed changes and the WSOD should now be addressed. Would love to hear from @david-szabo97 @Copons @Addison-Stavlo @vindl if there are any concerns. I moved the logic to handle conditional rendering of the Left Sidebar up to its parent, `edit-site/editor`

**Comment by jasmussen** [comment]
_Stellar_ work, @jeyip, I'm seeing this, and it's a substantial improvement:

featured

Thanks so much, I hope we can land this soon!

**Comment by david-szabo97** [comment]
> I pushed changes and the WSOD should now be addressed. Would love to hear from @david-szabo97 @Copons @Addison-Stavlo @vindl if there are any concerns. I moved the logic to handle conditional rendering of the Left Sidebar up to its parent, `edit-site/editor`

Any reason to move it to the parent? I prefer it to be in a separate component so less clutter in the `editor`. Opening the Inserter will now re-render the whole `Editor` if it's moved to the parent.

**Comment by jeyip** [comment]
> Any reason to move it to the parent?

Good question!

It's because the `InterfaceSkeleton` component is designed in such a way that it expects us to pass in a false-y value when an interface element (like leftSidebar, header, footer, etc.) should not be rendered. 

https://github.com/WordPress/gutenberg/blob/7778a3630356c3a26c8e91a1bba98f7bacbb3e80/packages/interface/src/components/interface-skeleton/index.js#L87-L96

In edit-site, however, we had the left sidebar set up in such a way that it was _always_ truthy

https://github.com/WordPress/gutenberg/blob/7778a3630356c3a26c8e91a1bba98f7bacbb3e80/packages/edit-site/src/components/editor/index.js#L212-L219

This was leading to the unexpected behavior we were seeing with the mobile wsod. The `InterfaceSkeleton` was _always_ rendering a leftSidebar wrapper, even when the inserter wasn't "open". 

> Opening the Inserter will now re-render the whole Editor if it's moved to the parent.

Hmmm...this is a good point. I'll try and think about alternative approaches, but would also love to hear if you have any ideas @david-szabo97

**Comment by jasmussen** [comment]
The post editor handles this same problem, worth noting. So we can look there for inspiration. 

Not sure if better, but it can be handled with CSS of we have a class name to indicate whether the left sidebar is open or not.

But before we go down a rabbit hole it really seems worth it to compare with the post editor.

**Comment by jeyip** [comment]
> The post editor handles this same problem, worth noting. So we can look there for inspiration.

Great idea! I ended up comparing our implementation with the post editor's implementation, and it looks like the post editor itself would run into the same issue of unnecessary re-renders, although I might be missing something.

> Not sure if better, but it can be handled with CSS of we have a class name to indicate whether the left sidebar is open or not.

Also gave this some thought earlier, but I think we'd have to pass more information into the InterfaceSkeleton which would also cause unnecessary re-renders. 🤔

**Comment by jasmussen** [comment]
I wonder if such a skeleton refactor should block this PR from landing, given:

- the site editor is currently unusable on < 600px
- your approach is "no worse" than what's been shipping for many months in the post editor

In that vein, we could consider this a regression fix with an opportunity to refactor the skeletons for the benefit of both editors?

**Comment by jeyip** [comment]
Ah I should have clarified @jasmussen. I completely agree with your sentiments. I think, in the worst case scenario, we should move forward with the changes in this PR for the reasons you listed above. 

I was interested in hearing from David to see if he had ideas about solutions that would meet both criteria of:
- Avoiding a rerender of the entire editor
- Addressing the mobile wsod

**Comment by david-szabo97** [comment]
Since the implementation is identical to the Post Editor, I think we are good to go. I can't think of any easy way to fix the rerender for now.

We will definitely need to rethink the `InterfaceSkeleton` and `Editor`. I don't think they are going to be bottlenecks for a while but they are the root of the whole Gutenberg so it'd be great to fix them up a little bit to avoid all that unnecessary rerenders.

**Review by david-szabo97 — APPROVED** [review]
One thing I noticed, we can open the Navigation sidebar by clicking on the name of the current template then clicking on "Browse templates". This opens the sidebar but there is no way to close it.

The name of the template in the header also gets pushed and it is no longer visible. (If you resize while the sidebar is open you will see that the name of the template is showing up once your browser is larger.)

image

**After all,** since this PR has been going on for a while, we would have a good foundation once this PR is merged, then we can address the issues coming up.

**Comment by jasmussen** [comment]
Thank you for the review David! I would agree there's very ripe ground for refactoring this, and it should be done. 

> One thing I noticed, we can open the Navigation sidebar by clicking on the name of the current template then clicking on "Browse templates". This opens the sidebar but there is no way to close it.

This observation, I think, is one of the best reasons for moving ahead with this, even if it isn't optimal. This is the sort of bugfix that has to help inform the sidebar work, which we can only discover when it's in some state of functional.

I'm going to give this a rebase, and see if that helps the checks pass.

**Comment by jeyip** [comment]
I rebased master again

**Comment by jeyip** [comment]
All specs are now passing 👍 Going to go ahead and merge the PR.

**Pull request merged**
